### ISSUE TITLE
[New rule] Redundant async

### ISSUE THREAD

**Issue by incendial** [dart-code-checker/dart-code-metrics#113]
**Please describe what the rule should do:**
Checks for redundant `async`.
Cases where `async` is useful include:
* You are using `await`.
* You are returning an error asynchronously. `async` and then `throw` is shorter than return `Future.error(...)`.
* You are returning a value and you want it implicitly wrapped in a future. `async` is shorter than `Future.value(...)`.

**What category of rule is this? (place an "X" next to just one item)**

[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

Bad:
```dart
Future<void> afterTwoThings(Future<void> first, Future<void> second) async {
  return Future.wait([first, second]);
}
```

Good:
```dart
Future<void> usesAwait(Future<String> later) async {
  print(await later);
}

Future<void> asyncError() async {
  throw 'Error!';
}

Future<void> asyncValue() async => 'value';
```

**Are you willing to submit a pull request to implement this rule?**
Yes

**Comment by incendial** [comment]
Available in 4.19.0 🚀
